### ISSUE TITLE
Question 

### ISSUE THREAD

**Issue by Hugo-cell111** [Haochen-Wang409/U2PL#125]
Hi! In the code https://github.com/Haochen-Wang409/U2PL/blob/main/train_semi.py#L514, contra_loss needs to be adapted to the distributed setting (dist.all_reduce(contra_loss)), and I wonder what will happen if I delete this line of code? When do I need to write it? Thanks!

**Comment by Haochen-Wang409** [comment]
You can delete this line when using one GPU for training.

**Comment by Hugo-cell111** [comment]
Could I explain it in this way: the built-in loss functions such as nn.crossentropyloss don't need to use it, but the customized loss functions such as compute_contra_memobank_loss need to use dist.all_reduce in order to gather the tensors from all GPUs?

**Comment by Haochen-Wang409** [comment]
Maybe you are right, we did not try it.

**Comment by Haochen-Wang409** [comment]
Maybe you are right, we did not try it.
